src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -2250,43 +2250,55 @@ public static ulong ToUInt64(string? value, int fromBase)
         }
 
         // Convert the byte value to a string in base fromBase
-        public static string ToString(byte value, int toBase)
-        {
-            if (toBase != 2 && toBase != 8 && toBase != 10 && toBase != 16)
-            {
-                throw new ArgumentException(SR.Arg_InvalidBase);
-            }
-            return ParseNumbers.IntToString((int)value, toBase, -1, ' ', ParseNumbers.PrintAsI1);
-        }
+        public static string ToString(byte value, int toBase) =>
+            ToString((int)value, toBase);
 
         // Convert the Int16 value to a string in base fromBase
-        public static string ToString(short value, int toBase)
-        {
-            if (toBase != 2 && toBase != 8 && toBase != 10 && toBase != 16)
+        public static string ToString(short value, int toBase) =>
+            toBase switch
             {
-                throw new ArgumentException(SR.Arg_InvalidBase);
-            }
-            return ParseNumbers.IntToString((int)value, toBase, -1, ' ', ParseNumbers.PrintAsI2);
-        }
+                2 => value.ToString("b"),
+                8 => ToOctalString((ushort)value),
+                10 => value.ToString(CultureInfo.InvariantCulture),
+                16 => value.ToString("x"),
+                _ => throw new ArgumentException(SR.Arg_InvalidBase),
```

Comment:
The combination of `pattern switch` + `throw` might make the function non-inlinable, even for constant `toBase`. Probably worth checking and changing the implementation if that's the case.

Reply:
Fixed.